src/cdk-experimental/testing/testbed.ts

```diff
@@ -0,0 +1,217 @@
+ */
+export function getNativeElement(testElement: TestElement): Element {
+  if (testElement instanceof UnitTestElement) {
+    return testElement.element;
+  }
+
+  throw new Error('Invalid element provided');
+}
+
+/**
+ * Locator implementation for testbed.
+ * Note that, this locator is exposed for internal usage, please do not use it.
+ */
+export class UnitTestLocator implements Locator {
+  private _rootElement: TestElement;
+
+  constructor(private _root: Element, private _stabilize: () => Promise<void>) {
+    this._rootElement = new UnitTestElement(_root, this._stabilize);
+  }
+
+  host(): TestElement {
+    return this._rootElement;
+  }
+
+  async find(css: string, options?: Options): Promise<TestElement|null> {
+    await this._stabilize();
+    const e = getElement(css, this._root, options);
+    if (e === null) {
+      return null;
+    }
+    return new UnitTestElement(e, this._stabilize);
+  }
+
+  async findAll(css: string): Promise<TestElement[]> {
+    await this._stabilize();
+    const elements = this._root.querySelectorAll(css);
+    const res: TestElement[] = [];
+    for (let i = 0; i < elements.length; i++) {
+      res.push(new UnitTestElement(elements[i], this._stabilize));
+    }
+    return res;
+  }
+
+  async load<T extends ComponentHarness>(
+    componentHarness: ComponentHarnessType<T>, css: string,
+    options?: Options): Promise<T|null> {
+    const root = getElement(css, this._root, options);
+    if (root === null) {
+      return null;
+    }
+    const locator = new UnitTestLocator(root, this._stabilize);
+    return new componentHarness(locator);
+  }
+
+  async loadAll<T extends ComponentHarness>(
+    componentHarness: ComponentHarnessType<T>,
+    rootSelector: string): Promise<T[]> {
+    await this._stabilize();
+    const roots = this._root.querySelectorAll(rootSelector);
+    const res: T[] = [];
+    for (let i = 0; i < roots.length; i++) {
+      const root = roots[i];
+      const locator = new UnitTestLocator(root, this._stabilize);
+      res.push(new componentHarness(locator));
+    }
+    return res;
+  }
+}
+
+class UnitTestElement implements TestElement {
+  constructor(
+    readonly element: Element, private _stabilize: () => Promise<void>) {}
+
+  async blur(): Promise<void> {
+    await this._stabilize();
+    (this.element as HTMLElement).blur();
+    await this._stabilize();
+  }
+
+  async clear(): Promise<void> {
+    await this._stabilize();
+    if (!(this.element instanceof HTMLInputElement ||
+      this.element instanceof HTMLTextAreaElement)) {
+      throw new Error('Attempting to clear an invalid element');
+    }
+    this.element.focus();
+    this.element.value = '';
+    this.element.dispatchEvent(new Event('input'));
+    await this._stabilize();
+  }
+
+  async click(): Promise<void> {
+    await this._stabilize();
+    (this.element as HTMLElement).click();
+    await this._stabilize();
+  }
+
+  async focus(): Promise<void> {
```

Comment:
Dealing with focus can be a bit flaky in unit tests, because we're not guaranteed for the browser window itself to be focused which causes some browsers not to move focus.

Reply:
Yeah good point, added some more utilities to `@angular/cdk/testing` around focus/blur which should take care of it.